libs/compute_api/src/spec.rs

```diff
@@ -105,8 +106,27 @@ pub struct ComputeSpec {
     // updated to fill these fields, we can make these non optional.
     pub tenant_id: Option<TenantId>,
     pub timeline_id: Option<TimelineId>,
+
+    /// Pageserver information can be passed in three different ways:
+    /// 1. Here in `pageserver_connection_info`
+    /// 2. In the `pageserver_connstring` field.
+    /// 3. in `cluster.settings`.
+    ///
+    /// The goal is to use method 1. everywhere. But for backwards-compatibility with old
+    /// versions of the control plane, `compute_ctl` will check 2. and 3. if the
+    /// `pageserver_connection_info` field is missing.
+    ///
+    /// If both `pageserver_connection_info` and `pageserver_connstring`+`shard_stripe_size` are
+    /// given, they must contain the same information.
+    pub pageserver_connection_info: Option<PageserverConnectionInfo>,
+
     pub pageserver_connstring: Option<String>,
 
+    /// Stripe size for pageserver sharding, in pages. This is set together with the legacy
+    /// `pageserver_connstring` field. When the modern `pageserver_connection_info` field is used,
+    /// the stripe size is stored in `pageserver_connection_info.stripe_size` instead.
+    pub shard_stripe_size: Option<ShardStripeSize>,
```

Comment:
This wasn't used in the cplane side. I assume we don't want to introduce it now right?

Reply:
CC @erikgrinaker @hlinnaka

Reply:
Piece by piece :smiley_cat: 

1. Yes, the old code in cplane does set a pg setting `neon.stripe_size` but only if `shards[0].StripeSize != nil`. Otherwise, it is **not** set.
2. I think the "conflicting" bit here is a bit exaggerated, in the sense that it is passed in the spec directly, and then also as a part of the `PageserverConnectionInfo`. This member here directly in the spec will duplicate the pg setting `neon.stripe_size`, in my understanding.
3. I think we meant more like that it hasn't been a part of the `ComputeSpec` before.